WIP: Autonomous

sean-mcguire opened the pull request.

njbackhaus committed:
Added actions BUILD ERRORS, DO NOT EXPECT TO BE ABLE TO BUILD THIS

njbackhaus committed:
Began work on adding/fixing Auton Actions.

njbackhaus committed:
Worked on autonomous actions and completeted both cargo and grab hatch.

njbackhaus committed:
Merge remote-tracking branch 'origin/master' into autonomous

njbackhaus committed:
Fix errors in lift hatch place and cargo grab.
Finished .h for both drive

njbackhaus committed:
Finished Auton Actions and changed i to Action in each action

njbackhaus committed:
Uncapitalized the m_ things and made switch statement

COREProgrammers committed:
Created all auton routine files.
Commiting so Sean and Nathan can work together

sean-mcguire committed:
Actions.h commit for sean.

sean-mcguire committed:
Work In Progress

COREProgrammers committed:
WIP: Tweaked GrabHatchAction for it to work.

sean-mcguire committed:
Work In Progress: Same as last commit

COREProgrammers committed:
Finished Top Rocket Hatch Auton Routine

sean-mcguire committed:
Work In Progress: For Sean

sean-mcguire committed:
We created the last of the autonomous routines.

sean-mcguire committed:
Merge branch 'autonomous' of https://github.com/core2062/CORE2019 into autonomous

COREProgrammers committed:
Created last auton routines and moved new nodes to top

Julia-Pockat commented on src/Autonomous/Actions/GrabCargoAction.cpp:
```diff
@@ -5,14 +5,15 @@
-void GrabCargoAction::ActionInit(GrabCargo i) {
-    m_GrabCargoRequestedAction = i;
+void GrabCargoAction::ActionInit(GrabCargo Action) {
+    m_GrabCargoRequestedAction = Action;
```
Generally we would have m_grabCargoRequestAction with the first word after the m_ not be capital.

aidan-lincicum committed:
Made some changes to auton actions and CargoInNearSideBay.
Also made CargoNearAndMidSideBay

aidan-lincicum committed:
Added auton functionality

COREProgrammers committed:
Added auton routines and actions.

COREProgrammers committed:
.

COREProgrammers committed:
Fixed build errors

aidan-lincicum committed:
Made it so the lift raises to put the cargo in during auto

Julia-Pockat committed:
Changes for waypoint follower

Julia-Pockat committed:
Building changs for waypoint follower

Julia-Pockat committed:
Merge branch 'autonomous' of https://github.com/core2062/CORE2019 into autonomous

Julia-Pockat committed:
Made changes for the waypoint follower. Paths need to be added to all of
the autonomous routines, some of the switch statements need default cases
and many of the nodes in the autons need to be combined.

Julia-Pockat committed:
Finished adding waypoint follower code to the drive subsystem.
Should be ready to test once there are paths to test.

Julia-Pockat committed:
Added paths to each of the autonomous routines to fix compilation errors
Will need to make more for specific paths and will need to separate out
by starting position
List of auto routines that we are going to focus on needs to be narrowed
down

COREProgrammers committed:
Work In Progress

COREProgrammers committed:
Merged with Autonomous Branch

COREProgrammers committed:
Added more options for auton actions

COREProgrammers committed:
Work In Progress

Julia-Pockat committed:
Added more paths for the rocket positions

COREProgrammers committed:
Work In Progress

Julia-Pockat committed:
Added more paths for autonomous

COREProgrammers committed:
Merge branch 'autonomous' of https://github.com/core2062/CORE2019 into autonomous

COREProgrammers committed:
WIP
